desc/integrals/quad_utils.py

```diff
@@ -190,14 +184,67 @@ def leggauss_lob(deg, interior_only=False):
     return x, w
 
 
+def uniform(deg):
+    """Gauss-Chebyshev quadrature with uniformly spaced nodes.
+
+    Returns quadrature points xₖ and weights wₖ for the approximate evaluation
+    of the integral ∫₋₁¹ f(x) dx ≈ ∑ₖ wₖ f(xₖ).
+
+    Parameters
+    ----------
+    deg : int
+        Number of quadrature points.
+
+    Returns
+    -------
+    x, w : (jnp.ndarray, jnp.ndarray)
+        Shape (deg, ).
+        Quadrature points and weights.
+
+    """
+    # Define x = 2/π arcsin y and g : y ↦ f(x(y)).
+    #   ∫₋₁¹ f(x) dx = 2/π ∫₋₁¹ (1−y²)⁻⁰ᐧ⁵ g(y) dy
+    # ∑ₖ wₖ f(x(yₖ)) = 2/π ∑ₖ ωₖ g(yₖ)
+    # Given roots yₖ of Chebyshev polynomial, x(yₖ) is uniform in (-1, 1).
+    x = jnp.arange(-deg + 1, deg + 1, 2) / deg
```

Comment:
I am also confused as Rory, in the docstring, you mention Gauss-Chabyshev, but the function itself has nothing to do with that. Probably the way you use it in your implementation is related to that but if this is going to be a general integral utility function, I would prefer function docstrings to explain what actually this function is doing.

Reply:
Ok thanks, would ""Gauss-Chebyshev quadrature in transformed variable?" be better

These are the x(y_k) where y_k are Chebyshev roots. It performs a Gauss-Chebyshev quadrature in the transformed variable y. I want to mention this somewhere because uniform spacing is still ambigous as to how the endpoints are chosen (i.e. midpoints spacing, include endpoints) and the weights associated to points (i.e. in some cases points near boundaries are weighted less). Specifying it is a Gauss-Chebyshev quadrature makes that explicit.

Reply:
I am also fine with making this method private

Reply:
Yeah, "Gauss-Chebyshev quadrature in transformed variable" is better to distinguish it from the actual Gauss-Chebyshev quadrature. Making it private could also work, since this is not a stand-alone quadrature function but helps to create one